packages/rocketchat-lib/server/methods/robotMethods.js

```diff
@@ -0,0 +1,25 @@
+Meteor.methods({
+	'robot.modelCall'(model, method, args) {
+		check(model, String);
+		check(method, String);
+		if (!Meteor.userId()) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', {
+				method: 'robot.modelCall'
+			});
+		}
+		if (!RocketChat.authz.hasRole(Meteor.userId(), 'robot')) {
+			throw new Meteor.Error('error-not-allowed', 'Not allowed', {
+				method: 'robot.modelCall'
+			});
+		}
+		const m = RocketChat.models[model];
+
+		if (!m || !_.isFunction(m[method])) {
+			throw new Meteor.Error('error-invalid-method', 'Invalid method', {
+				method: 'robot.modelCall'
+			});
+		}
+		const cursor = RocketChat.models[model][method].apply(RocketChat.models[model], args);
+		return cursor && cursor.fetch && cursor.fetch();
```

Comment:
This will return `undefined` when calling a `findOne`.

The called method could return a object or undefined from a findOne result or a cursor from a find result.
So the cursor can be an object from a record or an object from a cursor or undefined.

If the result is and cursor, that is ok, the fetch is the desired method to execute to get the data.
If the result is and object from a findOne you will return a property called `fetch` as undefined or will throw and error trying to execute the property as a function.

Take a look at the old code and try to use the same logic here